src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable react/prop-types */
+import React, { useContext } from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, act } from '@testing-library/react';
+import useClickTracker from '.';
+import pidginData from './fixtureData/tori-51745682.json';
+import {
+  RequestContext,
+  RequestContextProvider,
+} from '#contexts/RequestContext';
+import {
+  ServiceContext,
+  ServiceContextProvider,
+} from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+import * as params from '#containers/ATIAnalytics/params';
+import * as beacon from '#containers/ATIAnalytics/beacon';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+const defaultProps = {
+  pageData: pidginData,
+  componentName: 'brand',
+};
+
+const WithContexts = ({ children }) => (
+  <RequestContextProvider
+    bbcOrigin="https://www.test.bbc.com"
+    pageType={STORY_PAGE}
+    isAmp={false}
+    service="pidgin"
+    pathname="/pidgin/tori-51745682"
+  >
+    <ServiceContextProvider service="pidgin">{children}</ServiceContextProvider>
+  </RequestContextProvider>
+);
+
+const TestComponentContainer = () => {
+  const clickRef = useClickTracker(defaultProps);
+  return (
+    <div>
+      <TestComponent ref={clickRef} />
+    </div>
+  );
+};
+
+const TestComponent = React.forwardRef((_, ref) => {
+  return (
+    <div id="test-component" ref={ref}>
```

Comment:
Interesting will this make the click tracking component track clicks on:
- the div?
- the link?
- the button?
- all of them?

The component probably needs the flexibility to be used on either a containing element such as a `div` or a lower level element such as an `<a>` or `<button>`.

Reply:
I see below you test for this to some extent, would we want the flexibility to track clicks on the containing `div` and seperately track clicks on a contained element like the `<a>` or the `<button>`? So seperate click refs for the `div` and the `<button>`/`<a>`? Perhaps a questions for @JonBeeb and maybe worth a test in here if it's necessary?

Reply:
Once a click event listener is attached to an element, any element which is a child of that parent element can be clicked and the listener will fire. In the ati request it specifies the actual element that has been clicked on, whether that be a button or an anchor 👍

Reply:
Ah okay, I don't think you'd need this but a slightly weird edge case I was consider was:
- You want track that a containing component has been clicked, e.g. a collapsible panel that reveals some more content (https://www.bbc.co.uk/gel/guidelines/accordion)
- You then want track if a button inside that revealed content has also been clicked independently

I don't think we have anything like this but just came to mind. With this current solution I suspect you could only track the collapsible panel had been clicked.

Reply:
In this case I think we need to consider using stopPropagation on the event.